lib/db.js

```diff
@@ -110,11 +111,25 @@ function streamDataToDB (stream, pg, counts) {
 
 function saveResultsToS3 (output, s3, bucketName, key, s3ChunkSize) {
   const stream = new DataStream();
-  stream
+
+  const finishing = stream
+    .catch(ignore)
+    .map(({status, ...measurement}) => measurement)
     .use(streamDataToS3, s3, bucketName, key, s3ChunkSize)
     .catch(e => output.raise(e));
 
-  output.whenEnd().then(() => stream.end()).catch(ignore);
+  const endStream = () => {
```

Comment:
I'm still seeing a timeout when running the following command

`AWS_ACCESS_KEY_ID=ID AWS_SECRET_ACCESS_KEY=KEY AWS_BUCKET_NAME=openaq-fetches SAVE_TO_S3=true node index.js`

there are no running adapters and looking at the db, there are no new measurements being inserted for a long time before timeout, so I wonder if this is still not getting called?

Reply:
Not this actually, but I have a hunch...

Reply:
Hi Joe - I found the issue in an implementation of `empty` method in scramjet. I should fix it by tomorrow morning your time.

Reply:
Hi again. Just to let you know, the issue I found seems more complex to solve because of conflicting requirements in couple different methods - so I'm working on this currently, but it's taking more than than anticipated.

Reply:
Hi Joe, the commit I just pushed should fix the issue.